### PR TITLE
Crash fix. Cosmetic fixes.

### PR DESCRIPTION
**Pull request by SoundSSGood**
Fixed crash when unequip commander artifacts (regression).

I also noticed that CGrowingArtifact is used only for commander artifacts. Perhaps it makes sense add the same "growing" property to the hero's artifacts?

### CODE REVIEW

File: lib/ArtifactUtils.cpp [hunk 1]
```diff
@@ -204,7 +204,7 @@ DLL_LINKAGE CArtifactInstance * ArtifactUtils::createArtifact(CMap * map, const
 			art = ArtifactUtils::createScroll(SpellID(spellID));
 		}
 	}
-	else //TODO: create combined artifact instance for random artifacts, just in case
```
Comment (SoundSSGood) [inline lib/ArtifactUtils.cpp:207]: Generation of combined artifacts on random maps does not match the behavior of OH3. I suggest removing this TODO.

File: lib/CArtHandler.cpp [hunk 2]
```diff
@@ -819,9 +819,6 @@ void CArtifactInstance::init()
 std::string CArtifactInstance::getDescription() const
 {
 	std::string text = artType->getDescriptionTranslated();
-	if (!vstd::contains(text, '{'))
-		text = '{' + artType->getNameTranslated() + "}\n\n" + text; //workaround for new artifacts with single name, turns it to H3-style
```
Comment (SoundSSGood) [inline lib/CArtHandler.cpp:823]: I did not find a single case of using this workaround. I suggest removing it.

File: lib/ArtifactUtils.cpp (hunk 1 above)
Comment (IvanSavenko) [reply lib/ArtifactUtils.cpp:207]: Perhaps add an assertion then? So if somebody uses this function for combined artifact he won't end up with broken state.

**Review by IvanSavenko — APPROVED** [review]

**Comment by misiokles** [comment]
Growing bonus for hero's artifacts is `"type" : "GROWS_WITH_LEVEL" ` and works fine in 1.2.x
https://github.com/vcmi-mods/wake-of-gods/blob/vcmi-1.2/content/config/artifactsWoG.json

How about creature artifacts in daily builds? They can be worn on/off without issues?
https://github.com/vcmi-mods/wake-of-gods/blob/vcmi-1.2/Mods/StackArts/Content/config/warlordBanners.json

**Comment by SoundSSGood** [comment]
> Growing bonus for hero's artifacts is `"type" : "GROWS_WITH_LEVEL" ` and works fine in 1.2.x

GROWS_WITH_LEVEL - It is not growing artifacts. It is hero's level related artifacts. Growing artifacts have their own level, which increases after each battle. I think in the future it will be possible to add other conditions for growth.

File: lib/ArtifactUtils.cpp (hunk 1 above)
Comment (SoundSSGood) [reply lib/ArtifactUtils.cpp:207]: Just checked. If allow in the generator, the combined artifacts will appear on the map. So, assertion is not needed.

**Pull request merged**